mentat/commands.py

```diff
@@ -162,3 +169,37 @@ def argument_names(cls) -> list[str]:
     @classmethod
     def help_message(cls) -> str:
         return "Remove files from the code context"
+
+
+class UndoCommand(Command, command_name="undo"):
+    async def apply(self, *args: str) -> None:
+        stream = SESSION_STREAM.get()
+        code_file_manager = CODE_FILE_MANAGER.get()
+        errors = code_file_manager.history.undo()
+        message = "\nUndo complete"
+        await stream.send(errors + message)
+
+    @classmethod
+    def argument_names(cls) -> list[str]:
+        return []
+
+    @classmethod
+    def help_message(cls) -> str:
+        return "Undo the last change made by Mentat"
+
+
+class ResetCommand(Command, command_name="reset"):
```

Comment:
My initial thought seeing this was that "reset" resets the context. What are your thoughts on having "undo-all" undo everything? More explicit and harder to mis-type

Reply:
Hmm, I do think that's clearer. I was planning on having something like /clear clear all context, but I agree it might be confusing. I'll change this to undo-all